DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DataStructures.Lists 
+{
+    public class CircularBuffer<T> : IEnumerable<T> , ICollection<T> where T : IComparable<T> {
```

Comment:
Inconsistent formatting, sometimes you place the `{` on the same line and sometimes you place it on a new line. Please place all `{`, `}` characters on new lines.